watcher_test.go

```diff
@@ -83,101 +79,78 @@ func TestUpdate(t *testing.T) {
 }
 
 func TestUpdateForAddPolicy(t *testing.T) {
-	e, w := initWatcher(t)
-	_ = w.SetUpdateCallback(func(s string) {
-		CustomDefaultFunc(
-			func(id string, params interface{}) {
-				t.Fatalf("method mapping error")
-			},
-		)(s, nil, func(ID string, params interface{}) {
-			if ID != w.options.LocalID {
-				t.Fatalf("instance ID should be %s instead of %s", w.options.LocalID, ID)
-			}
-			expected := fmt.Sprintf("%v", []string{"alice", "book1", "write"})
-			res := fmt.Sprintf("%v", params)
-			if expected != res {
-				t.Fatalf("instance Params should be %s instead of %s", expected, res)
-			}
-		}, nil, nil, nil)
-	})
+	wo := WatcherOptions{
+		IgnoreSelf: true,
+	}
+	e, w := initWatcherWithOptions(t, wo)
+	e2, w2 := initWatcherWithOptions(t, wo)
+
+	time.Sleep(time.Millisecond * 500)
 	_, _ = e.AddPolicy("alice", "book1", "write")
+	time.Sleep(time.Millisecond * 500)
+	if !reflect.DeepEqual(e2.GetPolicy(), e.GetPolicy()) {
+		t.Error("These two enforcers' policies should be equal")
```

Comment:
I think more detailed error description should be added like which method（``AddPolicy``）, what is expected and what is actually. Refer to the part you deleted

Reply:
I don't quite understand what you mean. The previous test only tested that the received information was equal with the expectation, but did not test the processing of the information. Our ultimate goal is to synchronize the state among multiple instances, right?

Reply:
I mean the policy of e and e2 should be printed out, which is convenient for troubleshooting

Reply:
ok, it makes sense

Reply:
add another test for checking ID, and add log of method and policy of e/e2

Reply:
@RainshawGao

Reply:
@RainshawGao is this resolved?

Reply:
@hsluoyz yes, I have added more logs.